mlir/lib/Transforms/Utils/InliningUtils.cpp

```diff
@@ -25,22 +25,36 @@
 
 using namespace mlir;
 
-/// Remap locations from the inlined blocks with CallSiteLoc locations with the
-/// provided caller location.
+/// Remap all locations from the inlined blocks with CallSiteLoc locations with
+/// the provided caller location.
 static void
 remapInlinedLocations(iterator_range<Region::iterator> inlinedBlocks,
                       Location callerLoc) {
   DenseMap<Location, Location> mappedLocations;
-  auto remapOpLoc = [&](Operation *op) {
-    auto it = mappedLocations.find(op->getLoc());
+  auto remapLoc = [&](Location loc) {
+    auto it = mappedLocations.find(loc);
```

Comment:
```suggestion
    auto [remapLoc, inserted] = mappedLocations.try_emplace(loc);
    if (inserted) {
    
    }
```
Can we try emplace directly to avoid the double search?

Reply:
Not really. This cache only avoids querying the attribute uniquer, which doesn't work once we actually need to create the CallSiteLoc.

Reply:
Nvm, added this.